### PR TITLE
Lots of things

### PR DESCRIPTION
**Pull request by j-f1**
This PR:

* changes the internal API to overwrite `module.exports` (i.e. `require("./server.js")` instead of `require("./server.js").Server`
* allows dynamic config and routing files (`config.js`, `routes.js`; if CoffeeScript is installed: `routes.coffee`)
* Uses ES6 features (the linter is set to error on any use of `var`)
* Simplifies the config loading
* moves the JS files to `src/`
* allows multiple instances of `library` to be used in parallel
* makes file access async

I’m willing to split this up into multiple PRs, if you’d like.

### CODE REVIEW

File: src/handlers.js [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+      PATCH: {},
+      DELETE: {},
+      OPTIONS: {}
+    };
+
+    let rawRoutes;
+    try {
+      rawRoutes = require(config.appRoot + '/routes');
+    }
+    catch (err) {}
```
Comment (ArtOfCode-) [inline src/handlers.js]: Is xo not complaining about an unhandled error here?

Comment (j-f1) [inline src/handlers.js]: Nope. XO seems happy.

**Comment by j-f1** [comment]
Everything in the demo app still works the same as before, except HTML is rendered on the 404 page.

**Comment by ArtOfCode-** [comment]
It wasn't before? Whoops.

**Comment by j-f1** [comment]
@ArtOfCode- Config and routes can still be JSON. `require('./file')` tries `file.js` then `file.json`. If `require('coffee-script/register')` is called, then the files can be CoffeeScript.

**Comment by ArtOfCode-** [comment]
Fair. I haven't had chance to run this or look thoroughly yet, so I did the release notes based on a brief look.

**Comment by j-f1** [comment]
I switched to `mz/fs` because it uses promises, and Promises are Better Than Callbacks™ for this usecase.

**Pull request merged**